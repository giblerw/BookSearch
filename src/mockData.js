export const emptySet = {
    currentQuery: 'default',
    listData: {},
    totalItems: 0,
    activeSort: 'relevance',
}

export const set1 = {
    items: [{
        volumeInfo: {
            title: 'Testament to Youth',
            authors: ['Vera Brittain'],
            description: `This classic memoir of the First World War is now a major motion picture starring Alicia Vikander and Kit Harington. Includes an afterword by Kate Mosse OBE. In 1914 Vera Brittain was 20, and as war was declared she was preparing to study at Oxford. Four years later her life - and the life of her whole generation - had changed in a way that would have been unimaginable in the tranquil pre-war era. TESTAMENT OF YOUTH, one of the most famous autobiographies of the First World War, is Brittain's account of how she survived those agonising years; how she lost the man she loved; how she nursed the wounded and how she emerged into an altered world. A passionate record of a lost generation, it made Vera Brittain one of the best-loved writers of her time, and has lost none of its power to shock, move and enthral readers since its first publication in 1933.`,
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=YSI4AgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
}

export const set2 = {
    items: [{
        volumeInfo: {
            title: 'Hoot',
            authors: ['Carl Hiaasen'],
            description: `Roy, who is new to his small Florida community, befriends a mysterious boy who introduces him to potty-trained alligators, burrowing owls, and sparkly-tailed poisonous snakes. A Newbery Honor Book, ALA Notable Book, & ALA Best Book for Young Adults. Reissue.`,
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=BTDSfx5bunwC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
};

export const set3 = {
    items: [{
        volumeInfo: {
            title: 'Testament to Youth',
            authors: null,
            description: '',
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=YSI4AgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
}

export const beforeSearch = {
    query: '',
    searched: false
}

export const afterSearch = {
    query: 'harry',
    searched: true
}