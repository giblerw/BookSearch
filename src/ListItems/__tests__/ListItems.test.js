import { mount } from '@vue/test-utils';
import Vue from 'vue';

import ListItemsComponent from '../index.vue';
import * as mockProps from '../../mockData';

function getRenderedText (Component, propsData) {
    const Constructor = Vue.extend(Component)
    const vm = new Constructor({ propsData: propsData }).$mount()
    return vm.$el.textContent
  }

describe('ListItems.vue', () => {

    it('renders the text correctly with different props', () => {
        
        expect(getRenderedText(ListItemsComponent, {
            data: mockProps.set1
        })).toEqual(` 1. Testament to Youth  Author(s): Vera Brittain Book Summary:  This classic memoir of the First World War is now a major motion picture starring Alicia Vikander and Kit Harington. Includes an afterword by Kate Mosse OBE. In 1914 Vera Brittain was 20, and as war was declared she was preparing to study at Oxford. Four years later her life - and the life of her whole generation - had changed in a way that would have been unimaginable in the tranquil pre-war era. TESTAMENT OF YOUTH, one of the most famous autobiographies of the First World War, is Brittain's account of how she survived those agonising years; how she lost the man she loved; how she nursed the wounded and how she emerged into an altered world. A passionate record of a lost generation, it made Vera Brittain one of the best-loved writers of her time, and has lost none of its power to shock, move and enthral readers since its first publication in 1933.`);
    
        expect(getRenderedText(ListItemsComponent, {
            data: mockProps.set2
        })).toEqual(` 1. Hoot  Author(s): Carl Hiaasen Book Summary:  Roy, who is new to his small Florida community, befriends a mysterious boy who introduces him to potty-trained alligators, burrowing owls, and sparkly-tailed poisonous snakes. A Newbery Honor Book, ALA Notable Book, & ALA Best Book for Young Adults. Reissue.`);
    })

    it('renders N/A for empty list data', () => {
        const wrapper = mount(ListItemsComponent, {
            propsData: { data: mockProps.set3 }
        });

        expect(wrapper.find('[class="book-authors"]').text())
        .toEqual('N/A');

        expect(wrapper.find('[class="book-description"]').text())
        .toEqual('N/A');
    })

    it('renders the expected HTML structure', () => {
        const wrapper = mount(ListItemsComponent, {
            propsData: { data: mockProps.set1 }
        })

        expect(wrapper.element).toMatchSnapshot();
    })
});