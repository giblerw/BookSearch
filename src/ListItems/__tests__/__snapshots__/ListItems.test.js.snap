// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItems.vue renders the expected HTML structure 1`] = `
<div>
  <dl
    id="book-list"
  >
    <div>
      <hr />
       
      <dt
        class="book-title"
      >
        1. 
        <b>
          <u>
            Testament to Youth
          </u>
        </b>
      </dt>
       
      <dd
        class="row book-data-content"
      >
        <div
          class="col-lg-2 book-cover-img"
        >
          <img
            alt="book-cover-img"
            src="http://books.google.com/books/content?id=YSI4AgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
          />
        </div>
         
        <div
          class="col-lg-2"
        >
          <dt>
            Author(s):
          </dt>
           
          <div
            class="book-authors"
          >
            <dd>
              Vera Brittain
            </dd>
          </div>
        </div>
         
        <div
          class="col-lg-8 book-description"
        >
          <dt>
            Book Summary:
          </dt>
           
          <dd
            class="book-description"
          >
             This classic memoir of the First World War is now a major motion picture starring Alicia Vikander and Kit Harington. Includes an afterword by Kate Mosse OBE. In 1914 Vera Brittain was 20, and as war was declared she was preparing to study at Oxford. Four years later her life - and the life of her whole generation - had changed in a way that would have been unimaginable in the tranquil pre-war era. TESTAMENT OF YOUTH, one of the most famous autobiographies of the First World War, is Brittain's account of how she survived those agonising years; how she lost the man she loved; how she nursed the wounded and how she emerged into an altered world. A passionate record of a lost generation, it made Vera Brittain one of the best-loved writers of her time, and has lost none of its power to shock, move and enthral readers since its first publication in 1933.
          </dd>
        </div>
      </dd>
    </div>
  </dl>
</div>
`;
